### ISSUE TITLE
Random Review questions:

### ISSUE THREAD

**Issue by Connor-Bowley** [mturja-vf-ic-bd/SlicerDeepLearningUI#2]
This is just a list of questions I would tend to put in a code review. Because they are not necessarily issues (I could be misunderstanding something) I am putting them all together here, but in separate comments. If any of them rise to level of an individual issue, we use the "Reference in new issue" feature to create a new issue for it.

**Comment by Connor-Bowley** [comment]
Are the default values for `FeatureNameLineEdit` and `TimePointLineEdit` actually good generic default values, or are theu specific to your particular training data?
https://github.com/mturja-vf-ic-bd/SlicerDeepLearningUI/blob/16810de8b7da5da2a08f41c548e79603c118241d/DeepLearner/DeepLearner.py#L83

**Comment by Connor-Bowley** [comment]
Is the choice to connect to signal `currentNodeChanged(vtkMRMLNode*)` for `TrainDirLineEdit, maxEpochLineEdit, batchSizeLineEdit, learningRateLineEdit` correct? I don't QLineEdit has the signal. Would `textChanged` be the better signal? [QLineEdit signals](https://doc.qt.io/qt-5/qlineedit.html#signals)

**Comment by Connor-Bowley** [comment]
In `initializeParameterNode`, it seems you are changing the return type (based on the commented out code) from None to int. In python, you have a function you want to do nothing, the `pass` keyword is typically used.

E.g.
```
def initializeParameterNode(self):
    pass
```

https://github.com/mturja-vf-ic-bd/SlicerDeepLearningUI/blob/16810de8b7da5da2a08f41c548e79603c118241d/DeepLearner/DeepLearner.py#L157-L171

**Comment by mturja-vf-ic-bd** [comment]
> 

These are dataset specific default values.

**Comment by Connor-Bowley** [comment]
> Do you suggest changing them?

If this is meant to be a general purpose deep learning module, then yes. I could reasonably see them being blank by default and need to be set by the user.

**Comment by mturja-vf-ic-bd** [comment]
> The docstring for `DeepLearnerLogic.process` needs to describe the parameters for args if anyone is expected to be able to use it.
> 
> It would probably be even better if it took separate (documented) keyword arguments rather than a dictionary.

Changed the argument of the process function which now takes each argument individually instead of a dictionary. Also added proper documentation for these arguments.

**Comment by mturja-vf-ic-bd** [comment]
> 

I think 'TimePointLineEdit' is generic. The FeatureNameLineEdit is somewhat generic in that the user might have multiple features for the same subject and may want to train on different features separately.

**Comment by mturja-vf-ic-bd** [comment]
> Is the choice to connect to signal `currentNodeChanged(vtkMRMLNode*)` for `TrainDirLineEdit, maxEpochLineEdit, batchSizeLineEdit, learningRateLineEdit` correct? I don't QLineEdit has the signal. Would `textChanged` be the better signal? [QLineEdit signals](https://doc.qt.io/qt-5/qlineedit.html#signals)

Thanks for pointing that out. I agree.

**Comment by Connor-Bowley** [comment]
Re: args to process

Looks good. My only couple comments:
1. it seems that the `gpus` argument should probably be a boolean and not an int, and possibly `use_gpu` would be a more apt name.
2. From first glance and reading its documentation, I have no clue what the `monitor` argument does, so maybe that documentation could be fleshed out more (or add a url to something if appropriate)

**Comment by Connor-Bowley** [comment]
>  think 'TimePointLineEdit' is generic. The FeatureNameLineEdit ...

I wasn't questioning the line edits, just their default values.

**Comment by Connor-Bowley** [comment]
A couple nitpick thing you don't need to change, but could be nice.

The `Maximum Epoch`, `Learning Rate`, and `Batch Size` are all numbers, so `QSpinBox`, `QDoubleSpinBox`, and `ctkSliderWidget` all seem more appropriate than a line edit for taking input.

**Comment by mturja-vf-ic-bd** [comment]
> Re: args to process
> 
> Looks good. My only couple comments:
> 
> 1. it seems that the `gpus` argument should probably be a boolean and not an int, and possibly `use_gpu` would be a more apt name.
> 2. From first glance and reading its documentation, I have no clue what to put for the `monitor` argument, so maybe that documentation could be fleshed out more (or add a url to something if appropriate)

Fixed these issues.

**Comment by mturja-vf-ic-bd** [comment]
> A couple nitpick thing you don't need to change, but could be nice.
> 
> The `Maximum Epoch`, `Learning Rate`, and `Batch Size` are all numbers, so `QSpinBox`, `QDoubleSpinBox`, and `ctkSliderWidget` all seem more appropriate than a line edit for taking input.

Fixed these issues.

**Comment by Connor-Bowley** [comment]
RE: monitor argument

That looks good, thanks. I think `validatin/precision` might be a typo though

**Comment by Connor-Bowley** [comment]
I got warnings when running the module.
```
PythonQt: QObject::connect() signal 'textChanged(str)' does not exist on QLineEdit
PythonQt: QObject::connect() signal 'textChanged(str)' does not exist on QSpinBox
PythonQt: QObject::connect() signal 'textChanged(str)' does not exist on QSpinBox
PythonQt: QObject::connect() signal 'textChanged(str)' does not exist on QDoubleSpinBox
```
Probably from 
https://github.com/mturja-vf-ic-bd/SlicerDeepLearningUI/blob/e45948d5d00428fee03827b09f19513a4f727da2/DeepLearner/DeepLearner.py#L96-L100

For the QLineEdit which does have a `textChanged` signal, you can connect like this:
```
self.ui.TrainDirLineEdit.textChanged.connect(self.updateParameterNodeFromGUI)
```

Similarly for the spinboxes you can do something like
```
self.ui.epochSpinBox.valueChanged.connect(self.updateParameterNodeFromGUI)
```

**Comment by Connor-Bowley** [comment]
If you run start trainer again after running, it does not reset the progress bar back down to 0.

**Comment by mturja-vf-ic-bd** [comment]
Fixed those issues.

**Comment by Connor-Bowley** [comment]
Closing as issues have been addressed.
